deploy/charts/cert-manager/templates/deployment.yaml

```diff
@@ -158,6 +158,20 @@ spec:
           resources:
             {{- toYaml . | nindent 12 }}
           {{- end }}
+          # LivenessProbe settings are based on those used for the Kubernetes
+          # controller-manager. See:
+          # https://github.com/kubernetes/kubernetes/blob/806b30170c61a38fedd54cc9ede4cd6275a1ad3b/cmd/kubeadm/app/util/staticpod/utils.go#L241-L245
+          livenessProbe:
+            httpGet:
+              host: 127.0.0.1
+              port: 10257
+              path: /livez
+              scheme: HTTP
```

Comment:
In Kubernetes this is `HTTPS`, but that would require some mechanism for generating the serving certificate....which I suppose we can do in a future PR.

Reply:
why does /livez require HTTPS?

Reply:
@wallrj I think we want to add this port to the ports section in the container
```
ports:
          - containerPort: 9402
            name: http-metrics
            protocol: TCP
+         - containerPort: 10257
+           name: livez
+           protocol: TCP
```

Reply:
I don't think the `ports` do anything. But it might be useful to give the user an idea of what ports are in use.
I agree that 127.0.0.1 seems strange, an external service should be able to query this endpoint too.

Reply:
Changed it to port 9403 (next one up from the metrics-server port and added a named port as you suggested.
It is useful. I can now do:

```
~$ kubectl port-forward -n cert-manager deploy/cert-manager http-healthz
Forwarding from 127.0.0.1:9403 -> 9403
Forwarding from [::1]:9403 -> 9403
Handling connection for 9403
```

```
$ curl http://localhost:9403/livez?verbose
[+]leaderElection ok
livez check passed
```